### DISCUSSION TITLE
Texter to ClickSend - messages array is empty

### DISCUSSION THREAD

**Discussion by 6502nmos** [symfony/symfony#52486] [Q&A]
I am trying to get the Click-send notifier/texter to work and it seems as if what is being submitted to the API is the wrong format and hence I get a 400 response.

Basic test controller:
```php
#[Route('/textmessage', name: 'textmessage', methods: ['GET'])]
public function textmessage(TexterInterface $texter): Response
{
    $sms = new SmsMessage('+44nnnnnnnn', 'Message Text');

    $sentMessage = $texter->send($sms);

    dd($sentMessage);
}
```

composer.json
```json
"symfony/click-send-notifier": "6.3.*",
"symfony/notifier": "6.3.*",
```
notifier.yaml
```yaml
framework:
    notifier:
        texter_transports:
            clicksend: '%env(CLICKSEND_DSN)%'
```
On the symfony profiler the https request that is sent by texter is shown as:
```
POST | https://rest.clicksend.com/v3/sms/send |  
-- | -- | --
[▼   "auth_basic" => [▼
     "name@almadenrhee.com"     "24B631EE-8745-A2C2-925F-nnnnnnnnnnnnnn"   ] 
  "json" =>
 [▼     "body" => "Message Text"   ] ]
```

Which I don't think matches the Click-Send API format which expects an array of 'messages' which then contain the one or more message details.

The response back is:
```
  "response_headers" => [▼
    "HTTP/2 400 "
    "date: Tue, 07 Nov 2023 14:56:39 GMT"
    "content-type: application/json"
    "server: protected"
    "cache-control: no-cache, private"
    "access-control-allow-origin: *"
    "x-upstream-latency: 66"
    "x-proxy-latency: 51"
  ]
  "response_content" => [▼
    "{"http_code":400,"response_code":"BAD_REQUEST","response_msg":"The messages array is empty.","data":null}"
  ]
```
Any ideas on what else needs setting up to get this to work?

Next I will be trying it with cURL but would really prefer to have this working as integrated with other channels under notifier.

Thanks

Mark

**Comment by 6502nmos** [comment]
The click send API works as advertised when using cURL. I've now created my own ClickSend Transport with an extended Notification class to call it from. All working well with Messenger. If you end up here then I suggest don't use the Symfony Click Send Notifier as it's not in any way correctly configured for their API.

[click-send-notifier](https://github.com/symfony/click-send-notifier)

Presumably the API has changed at some point and this repository is now out of date or I could have missed some part of the setup of course.

**Comment by BafS** [comment]
I have the same issue, the structure sent is not correct. 

Symfony is sending something like


```js
{\n
    "source": "php",\n
    "from": "xxxx",\n
    "body": "Hello World",\n
    "to": "+41xxxxxxx",\n
}
```

but it should be

```js
{\n
    "messages": [\n
        {\n
            "source": "php",\n
            "from": "xxxx",\n
            "body": "Hello World",\n
            "to": "+41xxxxxxx",\n
        }\n
    ]\n
}
```

I'll send a patch
